File: java/test/_trajpoly7.java
import java.util.Arrays;
import liba.trajpoly7;

class _trajpoly7 {
    public static void main(String[] args) {
        trajpoly7 obj = new trajpoly7(1, 0, 1, 0, 1, 0, 1, 0, 1);
        System.out.println(Arrays.toString(obj.c0()));
        System.out.println(Arrays.toString(obj.c1()));
        System.out.println(Arrays.toString(obj.c2()));
        System.out.println(Arrays.toString(obj.c3()));
        System.out.println(obj.pos(0.5) + "," + obj.vel(0.5) + "," + obj.acc(0.5) + "," + obj.jer(0.5));
    }
}

File: javascript/test/trajpoly7.js
#!/usr/bin/env node
var liba = require("../liba")
function assert(expr, info) {
    if (!expr) {
        throw Error("assertion failed: " + info);
    }
}
function do_leak_check() {
    for (const arg of arguments) {
        arg.delete()
    }
    if (liba.do_leak_check) {
        liba.do_leak_check()
    }
}
var ctx = new liba.trajpoly7(1, 0, 1, 0, 1, 0, 1, 0, 1)
assert(ctx.p)
assert(ctx.v)
assert(ctx.a)
assert(ctx.j)
console.log(ctx.pos(0.5), ctx.vel(0.5), ctx.acc(0.5), ctx.jer(0.5))
do_leak_check(ctx)
